File: CemQuintas/src/app/Models/responsavel.ts
import {Uf} from './uf';
import {Cidade} from './cidade';

export class Responsavel {
	private _id_responsavel: number;
	private _nome: string;
	private _telefone: string;
	private _pai: string;
	private _mae: string;
	private _endereco: string;
	private _id_uf: Uf;
	private _rg: string;
	private _data_rg: Date;
	private _id_cidade_naturalidade: Cidade;

	public get IdResponsavel(): number {
		return this._id_responsavel;
	}

	public set IdResponsavel(idResponsavel: number) {
		this._id_responsavel = idResponsavel;
	}

	public get Nome(): string {
		return this._nome;
	}

	public set Nome(nome: string) {
		this._nome = nome;
	}

	public get Telefone(): string {
		return this._telefone;
	}

	public set Telefone(telefone: string) {
		this._telefone = telefone;
	}

	public get Pai(): string {
		return this._pai;
	}

	public set Pai(pai: string) {
		this._pai = pai;
	}

	public get Mae(): string {
		return this._mae;
	}

	public set Mae(mae: string) {
		this._mae = mae;
	}

	public get Endereco(): string {
		return this._endereco;
	}

	public set Endereco(endereco: string) {
		this._endereco = endereco;
	}

	public get Uf(): Uf {
		return this._id_uf;
	}

	public set Uf(uf) {
		this._id_uf = uf;
	}

	public get Rg(): string {
		return this._rg;
	}

	public set Rg(rg: string) {
		this._rg = rg;
	}

	public get DataRg(): Date {
		if (this._data_rg && !(this._data_rg instanceof Date)) {
			this._data_rg = new Date(this._data_rg);
		}
		return this._data_rg; 
	}

	public set DataRg(dataRg: Date) {
		this._data_rg = dataRg;
	}

	public get Cidade(): Cidade {
		return this._id_cidade_naturalidade;
	}

	public set Cidade(cidade:Cidade) {
		this._id_cidade_naturalidade = cidade;
	}

}

File: CemQuintas.OR/Responsavel.cs
using System;
using Regisoft;
using System.Collections.Generic ;

namespace CemQuintas.OR
{
	/// <summary>
	/// Gerado por RSClass - Gerador de Classes
	/// </summary>
	[Serializable]
	public /*sealed*/ class Responsavel
	{
		

		
		#region Private Members		

		private long? _id_responsavel; 
		private string _nome; 
		private string _telefone; 
		private string _pai; 
		private string _mae; 
		private string _endereco; 
		private Uf _id_uf; 
		private string _rg; 
		private DateTime? _data_rg; 
		private Cidade _id_cidade_naturalidade; 		
		#endregion

		
		
		#region Constructor

		public Responsavel()
		{
			_id_responsavel = null; 
			_nome = null;
			_telefone = null;
			_pai = null;
			_mae = null;
			_endereco = null;
			_id_uf = null; 
			_rg = null;
			_data_rg = null; 
			_id_cidade_naturalidade = null; 
		}
		#endregion // End of Default ( Empty ) Class Constuctor

		
		
		#region Public Properties
			
		public virtual long? IdResponsavel
		{
			get
			{ 
				return _id_responsavel;
			}
			set
			{
				_id_responsavel = value;
			}

		}
			


		public virtual string Nome
		{
			get
			{ 
				return _nome;
			}

			set	
			{	
				if( value == null )
					throw new ExceptionRS("Informe 'Nome'");
				
				if(  value.Length > 60)
					throw new ExceptionRS("Valor ultrapassa limite em 'Nome'");					

				_nome = value;
			}
		}
			
		public virtual string Telefone
		{
			get
			{ 
				return _telefone;
			}

			set	
			{	
				if(  value != null &&  value.Length > 25)
					throw new ExceptionRS("Valor ultrapassa limite em 'Telefone'");					

				_telefone = value;
			}
		}
			
		public virtual string Pai
		{
			get
			{ 
				return _pai;
			}

			set	
			{	
				if(  value != null &&  value.Length > 60)
					throw new ExceptionRS("Valor ultrapassa limite em 'Pai'");					

				_pai = value;
			}
		}
			
		public virtual string Mae
		{
			get
			{ 
				return _mae;
			}

			set	
			{	
				if(  value != null &&  value.Length > 60)
					throw new ExceptionRS("Valor ultrapassa limite em 'Mae'");					

				_mae = value;
			}
		}
			
		public virtual string Endereco
		{
			get
			{ 
				return _endereco;
			}

			set	
			{	
				if(  value != null &&  value.Length > 100)
					throw new ExceptionRS("Valor ultrapassa limite em 'Endereco'");					

				_endereco = value;
			}
		}
			
		public virtual Uf IdUf
		{
			get
			{ 
				return _id_uf;
			}
			set
			{
				_id_uf = value;
			}

		}
			
		public virtual string Rg
		{
			get
			{ 
				return _rg;
			}

			set	
			{	
				if(  value != null &&  value.Length > 15)
					throw new ExceptionRS("Valor ultrapassa limite em 'Rg'");					

				_rg = value;
			}
		}
			
		public virtual DateTime? DataRg
		{
			get
			{ 
				return _data_rg;
			}
			set
			{
				_data_rg = value;
			}

		}
			
		public virtual Cidade IdCidadeNaturalidade
		{
			get
			{ 
				return _id_cidade_naturalidade;
			}
			set
			{
				_id_cidade_naturalidade = value;
			}

		}
			
		#endregion 
		
		
		#region Public Functions

		#endregion

		
		
		#region Equals And HashCode Overrides
		/// <summary>
		/// local implementation of Equals based on unique value members
		/// </summary>
		public override bool Equals( object obj )
		{
			if( this == obj ) return true;
			if( ( obj == null ) || ( obj.GetType() != this.GetType() ) ) return false;
			Responsavel castObj = (Responsavel)obj; 
			return ( castObj != null ) &&
				( this._id_responsavel == castObj.IdResponsavel );
		}
		
		/// <summary>
		/// local implementation of GetHashCode based on unique value members
		/// </summary>
		public override int GetHashCode()
		{
			
			int hash = 57; 
			hash = 27 * hash * _id_responsavel.GetHashCode();
			return hash; 
		}
		#endregion
		
	}
}
